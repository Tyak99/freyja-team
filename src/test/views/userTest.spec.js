import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import { TestQuestions } from '../../views/ModuleTest/ModuleTest';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const store = mockStore({ test: {} });
const props = {
  getModuleTest: () => {},
  isLoading: true,
  store,
  match: { params: { moduleId: '4' } },
};

const setUp = () => {
  const component = shallow(<TestQuestions {...props} />);
  return component;
};
describe('Question component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
    sinon.spy(TestQuestions.prototype, 'handleChange');
  });
  afterEach(() => {
    TestQuestions.prototype.handleChange.restore();
  });
  it('should render without errors', () => {
    expect(component).toMatchSnapshot();
  });
  it('ensure onchange is called for TestQuestions', (done) => {
    const event = { target: { name: 4, value: 'customer' } };
    const questions = component.find('TestQuestions');
    expect(questions.length).toBe(1);
    questions.simulate('change', event);
    expect(TestQuestions.prototype.handleChange.called).toBe(true);
    done();
  });
});
