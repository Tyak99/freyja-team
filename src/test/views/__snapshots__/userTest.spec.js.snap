// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component should render without errors 1`] = `ShallowWrapper {}`;
